src/pudl/transform/eia.py

```diff
@@ -1149,133 +1140,171 @@ def fix_balancing_authority_codes_with_state(
+                ),
+            ),
+        },
+        required_resource_keys={"dataset_settings"},
+        name=f"harvested_{entity}_eia",
+    )
+    def harvested_entity(context, **eia_transformed_dfs):
+        """Harvesting IDs & consistent static attributes for EIA entity."""
+        logger.info(f"Harvesting IDs & consistent static attributes for EIA {entity}")
+        eia_settings = context.resources.dataset_settings.eia
+        # Do some final cleanup and assign appropriate types:
+        eia_transformed_dfs = {
+            name: convert_cols_dtypes(df, data_source="eia")
+            for name, df in eia_transformed_dfs.items()
+        }
+
+        if entity == "utilities":
+            # Remove location columns that are associated with plants, not utilities:
+            for table, df in eia_transformed_dfs.items():
+                if "plant_id_eia" in df.columns:
+                    plant_location_cols = [
+                        "street_address",
+                        "city",
+                        "state",
+                        "zip_code",
+                    ]
+                    logger.info(f"Removing {plant_location_cols} from {table} table.")
+                    eia_transformed_dfs[table] = df.drop(
+                        columns=plant_location_cols, errors="ignore"
+                    )
 
-        harvesting(
+        entity_df, annual_df, col_dfs = harvesting(
             entity,
             eia_transformed_dfs,
-            entities_dfs,
             debug=context.op_config["debug"],
             eia860m=eia_settings.eia860.eia860m,
         )
-    _boiler_generator_assn(
-        eia_transformed_dfs,
-        eia923_years=eia_settings.eia923.years,
-        eia860_years=eia_settings.eia860.years,
-        debug=context.op_config["debug"],
-    )
-    # get rid of the original annual dfs in the transformed dict
-    remove = ["generators", "plants", "utilities", "boilers"]
-    for entity in remove:
-        eia_transformed_dfs[f"{entity}_eia860"] = eia_transformed_dfs.pop(
-            f"{entity}_annual_eia", f"{entity}_annual_eia"
+
+        # Apply standard PUDL data types to the new entity tables:
+        pkg = Package.from_resource_ids()
+        entity_res = pkg.get_resource(f"{entity}_entity_eia")
+        entity_df = apply_pudl_dtypes(entity_df, group="eia").pipe(entity_res.encode)
+        annual_res = pkg.get_resource(f"{entity}_eia860")
+        annual_df = apply_pudl_dtypes(annual_df, group="eia").pipe(annual_res.encode)
+
+        if entity == "plants":
+            # Post-processing specific to the plants entity tables
+            entity_df = _add_additional_epacems_plants(entity_df).pipe(_add_timezone)
+            annual_df = fillna_balancing_authority_codes_via_names(annual_df).pipe(
+                fix_balancing_authority_codes_with_state, plants_entity=entity_df
+            )
+
+        # Need to figure out how to return the col_dfs if debug is True
```

Comment:
@bendnorman It would be great if we could just always write the `col_dfs` to disk using the default IO Manager, but I'm not sure how to do that. I'm not sure how many of these dataframes there are or what they are named. I think there's kind of a lot of them. One for every combination of entity and harvested column? @cmgosnell is that right?

Reply:
I think `col_dfs` has dozens of dataframes in it, one for each harvested column.

Reply:
Yes, @zaneselvans  you've got it. For every `entity` the `cols_df` is a dictionary of harvested column names (keys) and dataframes (values). The dataframes contain a record for every instance of where a value showed up in the natives tables for each group of `id_cols`. And an `occurrence` and `consistency` column. Which is to say its a lot of dataframes.

This is super helpful when a) a column isn't consistent enough for the harvesting and thus is breaking the builds or b) a user pings us about why a particular harvested value is X instead of Y. Both rare occurrences but have come up consistently over pudl's life. lolol.

I guess it would be nice to be able to have them lying around if something goes wrong. But this happens so rarely that if we can easily generate them in a notebook I think that is fiiiiiiiiine and I would prefer that to spamming ourselves with these many dfs.

Reply:
Okay, there are hundreds of these little dataframes and we access them rarely so I don't think outputting them makes sense. We can use the debugging notebook if/when we need to access them. I'll also return an empty dictionary for `col_dfs` if not debug.